pkg/apis/metal3/v1alpha1/baremetalhost_types.go

```diff
@@ -104,6 +104,81 @@ type BMCDetails struct {
 	CredentialsName string `json:"credentialsName"`
 }
 
+// DiskType is used to specify the disk type for a logical disk, e.g. hdd or ssd.
+type DiskType string
+
+// Valid disk types for RAID configuration
+const (
+	HDD DiskType = "hdd"
+	SSD DiskType = "ssd"
+)
+
+// DiskInterfaceType is used to specify the interface for a logical disk.
+type DiskInterfaceType string
+
+// Valid interface types for RAID configuration
+const (
+	SATA DiskInterfaceType = "sata"
+	SCSI DiskInterfaceType = "scsi"
+	SAS  DiskInterfaceType = "sas"
+)
+
+// RAIDLevel type is used to specify the RAID level for a logical disk.
+type RAIDLevel string
+
+// Valid RAID levels
+const (
+	RAID0  RAIDLevel = "0"
+	RAID1  RAIDLevel = "1"
+	RAID2  RAIDLevel = "2"
+	RAID5  RAIDLevel = "5"
```

Comment:
as per comment below, let's keep only RAID 0,1, and 1+0

Reply:
Done. I keep only RAID O, 1 and 1+0